File: scripts/generate_index.js
const fs = require('fs');
const path = require('path');

const dataDir = path.join(__dirname, '..', 'data');
const outFile = path.join(dataDir, 'index.json');

fs.readdir(dataDir, (err, files) => {
  if (err) {
    console.error('Failed to read data directory', err);
    process.exit(1);
  }
  const jsonFiles = files.filter(f => f.toLowerCase().endsWith('.json') && f !== 'index.json');
  fs.writeFile(outFile, JSON.stringify(jsonFiles, null, 2), (err) => {
    if (err) {
      console.error('Failed to write index.json', err);
      process.exit(1);
    }
    console.log('Wrote', outFile, 'with', jsonFiles.length, 'entries');
  });
});

File: scripts/generate_index.py
import os
import json
from pathlib import Path


def main():
    repo_root = Path(__file__).resolve().parents[1]
    data_dir = repo_root / 'data'
    out_file = data_dir / 'index.json'

    files = [f.name for f in data_dir.iterdir() if f.is_file() and f.suffix.lower() == '.json' and f.name != out_file.name]
    files.sort()

    with open(out_file, 'w', encoding='utf-8') as fh:
        json.dump(files, fh, indent=2, ensure_ascii=False)

    print(f'Wrote {out_file} with {len(files)} entries')


if __name__ == '__main__':
    main()
